### DISCUSSION TITLE
saleor-dashboard 2.11.1 how to create new menu

### DISCUSSION THREAD

**Discussion by Batgerel53** [saleor/saleor#8550] [Q&A]
![dash1](https://user-images.githubusercontent.com/56902009/141047708-55a452f2-8995-4629-9a15-b2b123284975.PNG)
![dash2](https://user-images.githubusercontent.com/56902009/141047723-9444bfb9-1814-4bb4-bea6-d97670d3f9b1.PNG)
![saleor2](https://user-images.githubusercontent.com/56902009/141047733-73f56372-5db1-4788-8761-bebb8510ad4f.PNG)

how do solved this problem help me

**Comment by karolkielecki** [comment]
@Batgerel53 Hi! 

Firstly, we recommend using the latest Saleor 3.0 beta version. Secondly, we don't recommend modifying Saleor Dashboard. What exactly would you like to achieve? I mean, you can always fork and modify the code to your needs, but it will be difficult to stay up to date with the next releases.

**Reply by Batgerel53** [reply]
![1](https://user-images.githubusercontent.com/56902009/141225124-8d031c67-da2c-4102-9d55-1af0a00e10d8.PNG)
![2](https://user-images.githubusercontent.com/56902009/141225144-4bb54abb-7c15-4add-8d97-2e517632d085.PNG)
what's wrong Help me please

**Comment by Batgerel53** [comment]
Thanks a lot.   It is my package.json file  . How to write codegen instead?

**Reply by Batgerel53** [reply]
@karolkielecki

**Comment by Batgerel53** [comment]
Liststages_sages_edges_node [] does not exist.

**Comment by lastmaj** [comment]
in your "package.json" file there is a script called "build-types". You need to execute that script in order to generate types. Your schema.graphql needs to be updated too.
So, assuming you did not change the name of the container, execute the following : 
`docker exec -it saleor-platform_dashboard_1 npm run build-types`

replace saleor-platform_dashboard_1 with the name of the container running your dashboard application.

**Reply by Batgerel53** [reply]
![sdf](https://user-images.githubusercontent.com/56902009/141281408-6dbc800a-e9c5-47ff-b1e4-3eddbccbdea6.PNG)

**Reply by Batgerel53** [reply]
I did the npm run build-types command

**Reply by Batgerel53** [reply]
schema.graphql has been updated.

**Reply by Batgerel53** [reply]
![ыбөбыө](https://user-images.githubusercontent.com/56902009/141281941-a760b30c-35a8-4476-b3dd-16d6a36e5281.PNG)
this is queries file

**Reply by Batgerel53** [reply]
@lastmaj

**Reply by lastmaj** [reply]
it appears that you have a problem with docker; the "build-types" command did not run. check your docker installation and run it again until you have a proper output.

**Reply by Batgerel53** [reply]
okay thanks
